jsk_naoqi_robot/jsk_pepper_startup/apps/meeting/scripts/meeting.l

```diff
@@ -0,0 +1,204 @@
+    (send *ri* :speak "\\vct=120\\\\rspd=80\\おつかれさまでしたッ！\\rspd=100\\")
+    (send *ri* :wait-interpolation)
+    (send *ri* :hide-image)
+    (send *ri* :set-language "English")
+    ;;
+    (set-alist 'description "End Meeting" args)
+    :finished))
+
+(defun comment-func (args)
+  (let (count str (wait 600) tm)
+    (ros::ros-info "comment func, *count* = ~A" *count*)
+    (setq *last-tm* (ros::time-now))
+    ;; wait for 600 sec OR if someone set *coutinue* nil, return with :finished
+    (while (and (ros::ok)
+                *continue*
+                (< (setq tm (send (ros::time- (ros::time-now) *last-tm*) :to-sec)) wait))
+      (ros::ros-info "wait for move .. continue = ~A, time ~A < ~A (wait)" *continue* tm wait)
+      (ros::spin-once)
+      (ros::duration-sleep 1))
+    (unless *continue*
+      (return-from comment-func :finished))
+    (setq *last-tm* (ros::time-now))
+    ;;
+    (setq count (mod *count* (length *img-list*)))
+    ;;
+    (send *ri* :set-basic-awareness-enabled nil)
+    (send *ri* :set-background-movement-enabled nil)
+    (send *ri* :show-image (elt *img-list* count))
+    (send *ri* :speak (elt *comment-list* count))
+    (send *ri* :angle-vector-sequence (elt *pose-list* count) (elt *time-list* count))
+    (send *ri* :wait-interpolation)
+    ;;(send *ri* :angle-vector (send *pepper* :spread-hands-smally))
+    (send *ri* :set-basic-awareness-enabled t)
+    (send *ri* :set-background-movement-enabled t)
+    ;;
+    (incf *count*)
+    (setq str (elt *comment-list* count))
+    ;; trim // ... //
+    (while (setq term1 (position #\\ str :count 1) term2 (position #\\ str :count 2))
+      (setq str (concatenate string (subseq str 0 term1) (subseq str (+ term2 1)))))
+    (ros::ros-info "comment func,  count  = ~A, str =  ~A" count str)
+    (set-alist 'description str args)
+    (when *image*
+         ;; not to send \n ???
+        (set-alist 'image (remove #\newline (base64encode (send *image* :serialize))) args)) ;; should be compressed??
+    :next))
+
+
+(load "package://roseus_smach/src/state-machine-ros.l")
+(defun meeting-sm ()
+  (let (sm)
+    (setq sm
+          (make-state-machine
+           '((:start :started :comment)    ;; transitions (node transition node)
+             (:comment :next :comment)
+             (:comment :finished :end)
+             (:end :finished :goal)
+             )
+           '((:start    'start-func)       ;; node-to-function maps
+             (:end      'end-func)
+             (:comment  'comment-func)
+             )
+           '(:start)                    ;; initial node
+           '(:goal)                     ;; goal node
+           ))
+    (send sm :arg-keys 'description)
+    sm))
+
+;; create robot interface
+(unless (boundp '*ri*) (pepper-init))
+(objects (list *pepper*))
+
+;;
+(defun image_cb (msg)
+  (setq *image* msg))
+(ros::subscribe "/pepper_robot/camera/front/image_raw" sensor_msgs::Image #'image_cb)
```

Comment:
I think it might be better to use compressd image if compressd image is published. Or manually compress it before base64encoding.
In my local environment with usb_cam_node(size: 640x480), I got an error about something like too long string and euslisp crashed.
```lisp
...
/opt/ros/noetic/share/euslisp/jskeus/eus/Linux64/bin/irteusgl: ERROR th=0 too long string ;; Segment\
ation Fault.
;; in (write-byte (bin2base64 x) s)
;; You are still in a signal handler.
;;Try reset or throw to upper level as soon as possible.
;; code=1740538864 x=67be82c0 addr=10000
Fatal:
```

Reply:
in that case, how about subscribe compressed image  using
```
-(ros::subscribe "/pepper_robot/camera/front/image_raw" sensor_msgs::Image #'image_cb)
+(ros::subscribe "/pepper_robot/camera/front/image_rect_color/compressed" sensor_msgs::CompressedImage #'image_cb)
```
and receives like-> https://gist.github.com/k-okada/4d0dabe396a6af37ca0511d40b4e21e4#file-smach_to_mail-py-compressed-version

Reply:
>in that case, how about subscribe compressed image using
 and receives like-> [gist.github.com/k-okada/4d0dabe396a6af37ca0511d40b4e21e4#file-smach_to_mail-py-compressed-version](https://gist.github.com/k-okada/4d0dabe396a6af37ca0511d40b4e21e4#file-smach_to_mail-py-compressed-version)


Yes, I have the almost same solution about compressed version.